src/main/java/com/google/sps/data/Recommended.java

```diff
@@ -0,0 +1,74 @@
+package com.google.sps.data;
+
+import java.util.*;
+
+/*This class holds the TYDRecommended algorithm which will give the
+    user a recommended store to get their desired beverage from.
+*/
+public class Recommended {
+
+  public static Drink getDrink(String storeId, String drinkName) {
+    List<Drink> drinkList = DrinkDAO.getDrinksByStore(storeId);
+    for (Drink d : drinkList) {
+      if (d.getName().equals(drinkName)) {
+        return d;
+      }
+    }
+    return null;
+  }
+  // given a coffeeshop, return its tydrecommended score
+  public static double getScore(Drink d) {
+    int numComments = d.getNumRatings();
+    double avgRating = d.getRating();
+    double distanceMinutes = getDistanceMins();
+    double commentScore, ratingScore, distanceScore;
+    //      /30             /35         /35
+
+    // get commentScore
+    if (numComments >= 100) { // 100+
+      commentScore = 30;
+    } else if (numComments >= 75) { // 75-99
+      commentScore = 25;
+    } else if (numComments >= 50) { // 50-74
+      commentScore = 20;
+    } else if (numComments >= 25) { // 25-49
+      commentScore = 15;
+    } else {
+      commentScore = ((double) numComments * .6);
+    }
+
+    // get ratingScore
+    ratingScore = avgRating * 7;
+
+    // get distanceScore
+    if (distanceMinutes <= 4.0) { // 0-4 mins
+      distanceScore = 35;
+    } else {
+      distanceScore = 35 - distanceMinutes;
+    }
+
+    return (commentScore + ratingScore + distanceScore);
+  }
+  // this function returns the highest scored store from a list of stores
+  public static String getBestShop(List<String> listStoreIds, String inputBeverage) {
```

Comment:
Could we write some tests for getBestShop() and getScore()?

Reply:
Finished :)